AnnService/src/IndexSearcher/main.cpp

```diff
@@ -187,29 +188,31 @@ int Process(std::shared_ptr<SearcherOptions> options, VectorIndex& index)
 
             std::atomic_size_t queriesSent(0);
             std::vector<std::thread> threads;
-            auto func = [&]()
-            {
-                size_t qid = 0;
-                while (true)
-                {
-                    qid = queriesSent.fetch_add(1);
-                    if (qid < numQuerys)
-                    {
-                        auto t1 = std::chrono::high_resolution_clock::now();
-                        index.SearchIndex(results[qid]);
-                        auto t2 = std::chrono::high_resolution_clock::now();
-                        latencies[qid] = (float)(std::chrono::duration_cast<std::chrono::microseconds>(t2 - t1).count() / 1000000.0);
-                    }
-                    else
-                    {
-                        return;
-                    }
-                }
-            };
 
             auto batchstart = std::chrono::high_resolution_clock::now();
 
-            for (std::uint32_t i = 0; i < options->m_threadNum; i++) { threads.emplace_back(func); }
+            for (std::uint32_t i = 0; i < options->m_threadNum; i++) { 
+                threads.emplace_back([&, i] {
+                    Helper::SetThreadAffinity(i, threads[i], 0, 0);
```

Comment:
Should all calling code to SPTAG search, set it's affinity like this? Can we make some small doc or comment, explain the options and tradeoffs for SPTAG case?